src/components/Post/PostScreen.js

```diff
@@ -0,0 +1,155 @@
+import React, { useState } from 'react';
+import {
+  Text, Button, View, StyleSheet,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import { ScrollView } from 'react-native-gesture-handler';
+import firestore from '@react-native-firebase/firestore';
+import Post from './Post';
+import Comment from './Comment';
+
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+  },
+  welcomeMessage: {
+    textAlign: 'center',
+    textAlignVertical: 'center',
+    height: 50,
+    fontSize: 20,
+  },
+});
+
+function CommentLoader({ postID }) {
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [commentList, setCommentList] = useState([]);
+  let commentsData = [];
+
+  firestore().collection('comments')
+    .get()
+    .then((snapshot) => {
+      if (!snapshot.empty) {
+        snapshot.forEach((anotherSnapshot) => {
+          const commentData = ({ ...anotherSnapshot.data(), id: anotherSnapshot.id });
+          if ((anotherSnapshot.get('postId') === postID) && !(commentsData.includes(commentData))) {
+            commentsData = commentsData.concat(commentData);
+          }
+        });
+      }
+    })
+    .then(() => {
+      setCommentList(commentsData.sort((a, b) => {
+        const aMillis = a.createdAt.toMillis();
+        const bMillis = b.createdAt.toMillis();
+        if (aMillis < bMillis) {
+          return -1;
+        }
+        if (aMillis > bMillis) {
+          return 1;
+        }
+        return 0;
+      }).map((comment) => {
+        const date = comment.createdAt.toDate();
+        return (
+          <Comment
+            key={comment.id}
+            name={comment.username}
+            title={comment.title}
+            createdAt={date.toTimeString()}
+            date={date.toDateString()}
+          >
+            {comment.body}
+          </Comment>
+        );
+      }));
+    })
+    .catch((error) => {
+      setErrorMessage(error.message);
+    });
+  return (
+    <View style={styles.container}>
+      <ScrollView>
+        {errorMessage && <Text>{errorMessage}</Text>}
+        {commentList}
+      </ScrollView>
+    </View>
+  );
+}
+
+export default function PostsScreen({ navigation }) {
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [postsList, setPostsList] = useState([]);
+
+  firestore().collection('posts')
```

Comment:
we can do a similar thing here, both with pagination and sorting the results in order. for now, we can replace this line by doing:

`Firestore.collection('posts').orderBy('createdAt', 'desc')`

We will worry about pagination later.